dogfood/main.tf

```diff
@@ -250,22 +283,16 @@ resource "docker_volume" "home_volume" {
   }
 }
 
-locals {
-  container_name = "coder-${data.coder_workspace.me.owner}-${lower(data.coder_workspace.me.name)}"
-  registry_name  = "codercom/oss-dogfood"
-}
-data "docker_registry_image" "dogfood" {
-  name = "${local.registry_name}:latest"
+resource "null_resource" "update_trigger" {
+  triggers = {
+    always_run = "${timestamp()}"
+  }
 }
 
 resource "docker_image" "dogfood" {
-  name = "${local.registry_name}@${data.docker_registry_image.dogfood.sha256_digest}"
-  pull_triggers = [
-    data.docker_registry_image.dogfood.sha256_digest,
-    sha1(join("", [for f in fileset(path.module, "files/*") : filesha1(f)])),
-    filesha1("Dockerfile"),
-  ]
-  keep_locally = true
+  name          = "${local.jfrog_host}/docker/${local.registry_name}:latest"
+  pull_triggers = [null_resource.update_trigger.id]
+  keep_locally  = true
 }
```

Comment:
This was not working nicely with a private docker registry, so I had to resort to this hack,
I filed an issue with the docker provider.
https://github.com/kreuzwerker/terraform-provider-docker/issues/593

Reply:
This is also not working, and terraform is not pulling the latest image even though Artifactory has it.
There are open issues regarding this: https://github.com/kreuzwerker/terraform-provider-docker/issues/172

Reply:
We can skip pulling from the satisfactory in Terraform or use a new tag on each build. Probably short git sha.

Reply:
After talking to @Emyrk, we decided not to use Artifactory to fetch pull docker image.